### ISSUE TITLE
Some details about Hyperparameter Optimization?

### ISSUE THREAD

**Issue by ClinuxMDL** [HIPS/neural-fingerprint#2]
I am so interested in your method that I had to ask another following issue:

Hyperparameter Optimization
"To optimize hyperparameters, we used random search. The hyperparameters
of all methods were optimized using 50 trials for each cross-validation fold. The
following hyperparameters were optimized: log learning rate, log of the initial weight scale, the log
L2 penalty, fingerprint length, fingerprint depth (up to 6), and the size of the hidden layer in the
fully-connected network. Additionally, the size of the hidden feature vector in the convolutional
neural fingerprint networks was optimized."

Could you give me some suggestions about Hyperparameter Optimization? 
Or any empirical range about these parameters? 

Best regards,
YJ

**Comment by duvenaud** [comment]
Hi YJ,

Good question!

Here are the parameter ranges we used when running the experiments in the paper:

```
morgan_bounds = dict(fp_length      = [16, 1024],
                     fp_depth       = [1, 4],
                     log_init_scale = [-2, -6],
                     log_step_size  = [-8, -4],
                     log_L2_reg     = [-6, 2],
                     h1_size        = [50, 100],
                     conv_width     = [5, 20])

neural_bounds = dict(fp_length      = [16, 128],   # Smaller upper range.
                     fp_depth       = [1, 4],
                     log_init_scale = [-2, -6],
                     log_step_size  = [-8, -4],
                     log_L2_reg     = [-6, 2],
                     h1_size        = [50, 100],
                     conv_width     = [5, 20])
```

I'd say that depth 2 or 3 is usually ok, and in general the middle of these ranges is a good place to start.  There was a lot of variability in the hypers that cross-validation chose, but that's probably because we only used 50 points in each fold.

Hope this helps!
